ast.ts

```diff
@@ -24,6 +24,8 @@ export type Scope<A> =
 
 export type Parameter = { name: string; type: Type; value?: Literal };
 
+export type Argument<A> = { a?: A; name?: string; value: Expr<A> }
```

Comment:
I don't think argument type is used anywhere. You should remove this

Reply:
Good point; we separate keyword arguments and arguments explicitly now so we don't need it anymore!